### ISSUE TITLE
loads of AHK application icons in task bar wen de- and re-activating hotkeys

### ISSUE THREAD

**Issue by nnako** [spyoungtech/ahk#216]
somehow related to #211 , when working with the following functions in a cycle, more and more task bar icons containing the AHK logo appear:
- `add_hotkey( ... )`
- `start_hotkeys()`
- `stop_hotkeys()`
- `clear_hotkeys()`

please see the small video which demonstrates the effect:

https://github.com/spyoungtech/ahk/assets/5845602/3394c0c0-2974-4e03-aeb6-7f14f34ff5ff

the source code is here:

```python
import time
from ahk import AHK

# globals
ahk = AHK()
request_quit = False


#
# callback function definitions
#

def on__ctrl_shift_alt_t():
    global ahk
    global request_quit
    print("CTRL+SHIFT+ALT+T pressed")

    # give some time for the control keys to reset
    time.sleep(0.2)

    # output keypresses
    ahk.send('__TEST__')
    ahk.send('{enter}')
    ahk.send('{F7}')

    # request quit of application
    request_quit = True

def on__f6():
    global ahk
    print("F6 pressed")

    # forward this key to active application
    ahk.active_window.send('{F6}')


#
# configurations and operation
#

while not request_quit:

    # initialize hotkeys
    ahk.add_hotkey("^+!t", on__ctrl_shift_alt_t)
    ahk.add_hotkey("f6", on__f6)
    ahk.start_hotkeys()

    # wait
    time.sleep(0.5)

    # de-initialize hotkeys
    ahk.stop_hotkeys()
    ahk.clear_hotkeys()

    # wait
    time.sleep(0.5)


# reset hotkeys
ahk.stop_hotkeys()
ahk.clear_hotkeys()

# grant some time to flush any pipeline
time.sleep(1)
```

**Comment by spyoungtech** [comment]
Yeah, this is a known behavior. As you noticed, the icons disappear when you mouse over them when Windows realizes the process is no longer running for those icons. So, there aren't actually that many processes actually running -- it's just the icon that's hanging around.

I'm _guessing_ this happens because the current process for restarting the process is not very nice and goes straight to `kill` which doesn't give the AHK process the opportunity to gracefully exit and remove the icon on its own (not sure if it will do that anyhow, though). I'm not sure if there's a more graceful way to tell the subprocess to exit here. But I haven't yet had an opportunity to investigate this issue.

It might be possible send some signal other than kill/terminate to stop the process (maybe this requires implementing signal handling in AHK?).

As a workaround, maybe hiding the tray icon altogether would work for you. Right now, there is no interface for this with respect to the process used for hotkeys, but is planned as part of #211

**Comment by spyoungtech** [comment]
FYI: a release candidate `1.2.0rc2` is available implementing the feature request in #211 

If all works out, it should be included in the upcoming 1.2.0 release, which should be forthcoming.

**Comment by spyoungtech** [comment]
1.2.0 is now released with the ability to hide the tray icon for hotkeys.
